### ISSUE TITLE
Strict handling of request options?

### ISSUE THREAD

**Issue by ChrisBeaumont** [Paradigm4/shim#26]
I spent some time trying to debug why compressed transfer wasn't working, until I realized that the version of shim I was using predated compression (and hence the compression related parameters I was adding to the URL were being ignored). I would have caught this sooner if shim had complained about encountering an unrecognized keyword. This might be useful for several such user errors.

**Comment by bwlewis** [comment]
I agree it's a good idea to return an error on unknown query parameters. I'll add this.
